test/adaptive_load/fake_plugins/fake_step_controller/fake_step_controller.proto

```diff
@@ -5,12 +5,23 @@ package nighthawk.adaptive_load;
 import "envoy/config/core/v3/extension.proto";
 import "google/rpc/status.proto";
 
-// Configuration for FakeStepController (plugin name: "nighthawk.fake_step_controller") that always
-// returns a fixed RPS value and changes converged and doomed states based on the latest reported
-// BenchmarkResult.
+// Configuration for FakeStepController (plugin name: "nighthawk.fake_step_controller") that returns
+// a fixed RPS value and changes converged and doomed states based on the latest reported
+// BenchmarkResult. Can also be programmed to return a proto validation failure, return an error
+// from input value setting every time, or return an error after some number of UpdateAndRecompute()
+// iterations.
 message FakeStepControllerConfig {
-  // RPS that should always be returned. Optional, default 0.
+  // RPS that should always be returned, except when artificial errors are configured. Optional,
+  // default 0.
   int32 fixed_rps_value = 1;
   // Artificial error that the plugin factory should return during validation. Optional.
   google.rpc.Status artificial_validation_failure = 2;
+  // Artificial error that should be returned from GetCurrentCommandLineOptions(). Optional.
+  google.rpc.Status artificial_input_setting_failure = 3;
```

Comment:
I think I'm having trouble understanding the full use case here.  If this is 3, then we are going to succeed 3 times, then fail on the 4th attempt. Makes sense, but I'm not sure I understand why that's useful.  What is the test that you're supporting by creating it in this way?

Reply:
`GetCurrentCommandLine()` can return an error status that should be handled cleanly by the main controller loop. The controller calls `GetCurrentCommandLine()` repeatedly during the adjusting stage until the step controller says it's converged, and then in the testing stage we call `GetCurrentCommandLine()` one last time, reusing the last converged value. 

In order to test handling of these errors, we need the `FakeStepController` to return successfully from `GetCurrentCommandLine()` during the adjusting stage, but then start returning errors just in time for the testing stage. We don't have any way to update the `FakeStepController` during the run, so it has to somehow be programmed up front to behave differently at different times.

An alternative would be for magic values in `UpdateAndRecompute()` to trigger `GetCurrentCommandLine()` error behavior. We already use magic values to control convergence and doom. But there's only so much information we can encode in metric score `double`s without having it get out of hand. Currently the `UpdateAndRecompute()` behavior is: zero scores=non-converged non-doomed, any positive score=converged, any negative score=doomed.

This trick wouldn't be necessary if the step controller were aware of what stage it was operating in.